src/main/java/io/trino/plugin/tiledb/TileDBClient.java

```diff
@@ -283,13 +270,24 @@ public Config buildConfig()
         }
     }
 
-    public Context buildContext(ConnectorSession session) throws TileDBError
+    public Context buildContext(ConnectorSession session, EncryptionType encryptionType, String encryptionKey) throws TileDBError
     {
-        Context localCtx = ctx;
+        Config tileDBConfig = new Config();
+        boolean updateCtx = false;
+
+        // Encryption
+        String currentKey = ctx.getConfig().get("sm.encryption_key");
+        boolean currentDifferent = !currentKey.equals(encryptionKey) && currentKey.length() > 0;
+        // add a key only if there is currently none or the current is different
+        if (encryptionType != null && encryptionKey != null
+                && (currentKey.equals("") || currentDifferent)) {
```

Comment:
I think `currentKey.equals("")` might be redundant with `currentDifferent`?

Reply:
oops!